### PR TITLE
Renewal hateffect update

### PR DESCRIPTION
**Pull request by attackjom**
* **Addressed Issue(s)**: 

* **Server Mode**:  Renewal

* **Description of Pull Request**: 

   Update new hat effect and some costume

### CODE REVIEW

**Comment by siriousb** [comment]
digital space not work.

**Comment by attackjom** [comment]
> digital space not work.

It work when use in client above ragexe20180621RE

**Comment by cahya1992** [comment]
so if i use ragexe20180621 will not working sir?

**Comment by attackjom** [comment]
> so if i use ragexe20180621 will not working sir?

Yes, for "Digital Space effect"

**Comment by cahya1992** [comment]
> > so if i use ragexe20180621 will not working sir?
> 
> Yes, for "Digital Space effect"

So sad :(

**Comment by cahya1992** [comment]
hateffet "HAT_EF_DOUBLEGUMGANG" not working too in 20180621

**Comment by attackjom** [comment]
My test client. (ragexe20180919a)
image

**Comment by cahya1992** [comment]
its use rathena emulator?

**Comment by attackjom** [comment]
> its use rathena emulator?

yes rathena can use perfectly until 20180919ragexe (not RE)

**Comment by OptimusM** [comment]
@attackjom im using the same client version (just for testing on my test server) and still missing some of those animations, like Digital Space and others

**Comment by attackjom** [comment]
> @attackjom im using the same client version (just for testing on my test server) and still missing some of those animations, like Digital Space and others

Do you have a newest hateffectinfo.lub files in you data ?

**Comment by attackjom** [comment]
> @attackjom im using the same client version (just for testing on my test server) and still missing some of those animations, like Digital Space and others

check your hateffect.lub  files

File: src/map/script.hpp [hunk 1]
```diff
@@ -1895,6 +1895,43 @@ enum e_hat_effects {
 	HAT_EF_SUBJECT_AURA_WHITE,
 	HAT_EF_SUBJECT_AURA_RED,
 	HAT_EF_C_SHINING_ANGEL_WING,
+	HAT_EF_MAGIC_STAR_TW,
+	HAT_DIGITAL_SPACE,
```
Comment (aleos89) [inline src/map/script.hpp]: I know this is a copy from kRO, but do you mind adjusting this to `HAT_EF_DIGITAL_SPACE`. Don't forget to update the export and item as well.

Comment (attackjom) [reply src/map/script.hpp]: sure i will fix later

**Comment by violent01** [comment]
> > @attackjom im using the same client version (just for testing on my test server) and still missing some of those animations, like Digital Space and others
> 
> Do you have a newest hateffectinfo.lub files in you data ?

where can i find the newest hateffectinfo.lub?

**Comment by attackjom** [comment]
Zackdreaver repository

File: src/map/script.hpp [hunk 2]
```diff
@@ -1895,6 +1895,43 @@ enum e_hat_effects {
...
+	HAT_EF_SLEIPNIR,
+	HAT_EF_C_MAPLE_WHICH_FALLS_RD,
+	HAT_EF_MAGICCIRCLERAINBOW,
+	HAT_EF_SNOWFLAKE_TIARA,
+	HAT_EF_MIDGARTS_GLORY,
+	HAT_EF_LEVEL99_TIGER,
+	HAT_EF_LEVEL160_TIGER,
+	HAT_EF_FLUFFYWING,
+	HAT_EF_C_GHOST_EFFECT,
+	HAT_EF_C_POPPING_PORING_AURA,
```
Comment (Angelic234) [inline src/map/script.hpp]: You miss HAT_EF_ResonateTaego, (item id 19433's hateffect)

File: src/map/script_constants.hpp [hunk 3]
```diff
@@ -7105,6 +7105,43 @@
...
+	export_constant(HAT_EF_SLEIPNIR);
+	export_constant(HAT_EF_C_MAPLE_WHICH_FALLS_RD);
+	export_constant(HAT_EF_MAGICCIRCLERAINBOW);
+	export_constant(HAT_EF_SNOWFLAKE_TIARA);
+	export_constant(HAT_EF_MIDGARTS_GLORY);
+	export_constant(HAT_EF_LEVEL99_TIGER);
+	export_constant(HAT_EF_LEVEL160_TIGER);
+	export_constant(HAT_EF_FLUFFYWING);
+	export_constant(HAT_EF_C_GHOST_EFFECT);
+	export_constant(HAT_EF_C_POPPING_PORING_AURA);
```
Comment (Angelic234) [inline src/map/script_constants.hpp]: export_constant(HAT_EF_ResonateTaego);
The same reason.

Comment (attackjom) [reply src/map/script_constants.hpp]: done

File: db/re/item_db.txt [hunk 4]
```diff
@@ -12624,6 +12638,9 @@
 31476,C_Cherry_Blossom_Hat_YL,C Yellow Cherry Blossom Hat,4,0,,0,,0,,0,0xFFFFFFFF,63,2,1024,,1,0,1727,{},{},{}
 31478,C_Eleanor_Wig_,C Eleanors Wig Lower,4,0,,0,,0,,0,0xFFFFFFFF,63,2,4096,,1,0,1502,{},{},{}
 //
+31479,C_Group_of_Stars,Costume Group of Stars,4,,,0,,,,,0xFFFFFFFF,63,2,4096,,1,0,1729,{ hateffect(HAT_EF_MANYSTARS,true); },{},{ hateffect(HAT_EF_MANYSTARS,false); }
+31588,C_Snow_Flower,C Snow Flower,4,0,,,,0,,0,0xFFFFFFFF,63,2,1024,,1,0,1799,{ hateffect(HAT_EF_SNOWFLAKE_TIARA,true); },{},{ hateffect(HAT_EF_SNOWFLAKE_TIARA,false); }
+31602,C_Midgarts_Glory,C Rune Midgarts Glory,,0,,,,0,,0,0xFFFFFFFF,63,2,1024,,1,,1807,{ hateffect(HAT_EF_MIDGARTS_GLORY,true); },{},{ hateffect(HAT_EF_MIDGARTS_GLORY,false); }
```
Comment (Angelic234) [inline db/re/item_db.txt]: Type should be 12(Shadow Equipment) not 0(Healing item)

**Comment by iubantot** [comment]
> My test client. (ragexe20180919a)
> image

may i ask for help about my problem im trying to test 2018 09 19 [Link to rA](https://rathena.org/board/topic/119025-2018-09-19-failed-to-connect-to-server/#comment-360817)

File: db/re/item_db.txt (hunk 4 above)
Comment (aleos89) [reply db/re/item_db.txt]: This was fixed [here](884043b).

**Pull request merged**